### PR TITLE
Issue/64 http client

### LINKED ISSUE #64: implement http-client

### PR DESCRIPTION
**Pull request by diehmlukas**
NOTE: a adapter for the config is used

closed #64

### CODE REVIEW

File: app/lib/api/client.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  if (response.statusCode == 200) {
+    return response.body;
+  } else {
+    final int statusCode = response.statusCode;
+    throw LoadConfigException('Failed to load config. \n'
+        'Response status code: $statusCode');
+  }
+}
+
+/// Send the full database to the Server.
```
Comment (AnnaGsell) [inline app/lib/api/client.dart:40]:
```suggestion
/// Sends the full database to the Server.
```

File: app/lib/api/client.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    },
+    body: jsonDatabase,
+  );
+  if (response.statusCode != 200) {
+    final int statusCode = response.statusCode;
+    throw SendDataException('Failed to send data.\n'
+        'Response status code: $statusCode');
+  }
+
+//ignore: avoid_print
```
Comment (mac641) [inline app/lib/api/client.dart:57]: Add TODO why this print needs to stay

File: app/lib/config/config_manager.dart [hunk 3]
```diff
@@ -67,11 +68,19 @@ class ConfigManager {
     }
   }
 
-// TODO: write doc comments
-// TODO: implement based on server API
-// void fetchConfigFromServer() async {
-//   _clientConfig = ... json object from server
-// }
+  /// This method loads the config from a server, and write it to [_clientConfig].
```
Comment (AnnaGsell) [inline app/lib/config/config_manager.dart:71]:
```suggestion
  /// This method loads the config from a server, and writes it to [_clientConfig].
```

File: app/lib/exceptions/exceptions.dart [hunk 4]
```diff
@@ -32,3 +32,13 @@ class MalformedServerUrlException implements Exception {
   final String msg;
   MalformedServerUrlException(this.msg);
 }
+
+class LoadConfigException implements Exception {
```
Comment (mac641) [inline app/lib/exceptions/exceptions.dart:36]:
```suggestion
class HttpErrorCodeException implements Exception {
```

File: app/lib/exceptions/exceptions.dart [hunk 5]
```diff
@@ -32,3 +32,13 @@ class MalformedServerUrlException implements Exception {
...
+  final String msg;
+  LoadConfigException(this.msg);
+}
+
+class SendDataException implements Exception {
+  final String msg;
+  SendDataException(this.msg);
+}
```
Comment (mac641) [inline app/lib/exceptions/exceptions.dart:44]: Remove this and replace all occurrences with `HttpErrorCodeException`

File: server/cmd/server.go [hunk 6]
```diff
@@ -139,3 +142,15 @@ func data(w http.ResponseWriter, r *http.Request) {
...
+// 	_, _ := io.ReadAll(r.Body)
+// 	io.WriteString(w, "it's a config for you")
+// 	fmt.Println("replyed config")
+// }
+//
+// func data(w http.ResponseWriter, r *http.Request) {
+// 	io.WriteString(w, "thanks for the data")
+//
+// 	fmt.Println("recieved data")
+// }
```
Comment (AnnaGsell) [inline server/cmd/server.go:156]: Please remove unused code ;)

File: server/cmd/server.go [hunk 7]
```diff
@@ -53,7 +53,6 @@ func serveRun(cmd *cobra.Command, args []string) {
 	// Set routing rules
```
Comment (AnnaGsell) [inline server/cmd/server.go:53]: This is old server.go code. Please remove this file from your pr :)
use
`git checkout main -- server/cmd/server.go`

File: app/test/config_manager_test.dart [hunk 8]
```diff
@@ -51,6 +55,7 @@ final Map<String, dynamic> customAssertObject = <String, dynamic>{
   'questions': _serializeQuestionnaireObjects(questions)
 };
 
+@GenerateMocks([Client])
```
Comment (mac641) [inline app/test/config_manager_test.dart:58]:
```suggestion

```

Comment (mac641) [reply app/test/config_manager_test.dart:58]: This is no longer needed since `fetchConfigFromServer` is not being tested

**Review by mac641 — CHANGES_REQUESTED** [review]
Please check my comments and apply these suggestions :)

**Review by AnnaGsell — CHANGES_REQUESTED** [review]

**Review by AnnaGsell — APPROVED** [review]

**Review by mac641 — APPROVED** [review]
Nice work 🎉

**Pull request merged**